src/system/OS_info/os_info_windows.cpp

```diff
@@ -80,10 +83,38 @@ static std::string version_name() {
 	return ret.substr(0, ret.find('|'));
 }
 
+unsigned int build_number() {
+	HKEY hkey{};
+	if(RegOpenKeyExA(HKEY_LOCAL_MACHINE, R"(Software\Microsoft\Windows NT\CurrentVersion)", 0, KEY_READ, &hkey))
+		return {};
+
+	DWORD ubr      = 0;
+	DWORD ubr_size = sizeof(ubr);
+	if(!RegQueryValueExA(hkey, "UBR", nullptr, nullptr, reinterpret_cast<LPBYTE>(&ubr), &ubr_size))
+		return ubr;
 
+	// Fall back to BuildLabEx in the early version of Windows 8.1 and less.
+	DWORD buildlabex_size = 0;
+	if(RegQueryValueExA(hkey, "BuildLabEx", nullptr, nullptr, nullptr, &buildlabex_size))
+		return {};
+
+	std::vector<char> buildlabex(buildlabex_size);
```

Comment:
std::string; REG_SZ is allowed to not be NUL-terminated and then strtok will smash whatever's after it